timed/projects/models.py

```diff
@@ -242,3 +247,18 @@ class TaskAssignee(models.Model):
     is_resource = models.BooleanField(default=False)
     is_reviewer = models.BooleanField(default=False)
     is_manager = models.BooleanField(default=False)
+
+
+@receiver(pre_save, sender=Project)
+def update_billed_flag_on_reports(sender, instance, **kwargs):
+    """Update billed flag on all reports from the updated project.
+
+    Only update reports if the billed flag on the project was changed.
+    """
+    # check whether the project was create or is being updated
+    if instance.pk:
+        if instance.billed != Project.objects.get(id=instance.id).billed:
+            for report in Report.objects.filter(Q(task__project=instance)):
+                if report.billed != instance.billed:
+                    report.billed = instance.billed
+                    report.save()
```

Comment:
```suggestion
            Report.objects.filter(Q(task__project=instance)).update(billed=instance.billed)
```

No need to iterate over every report - it's much easier to just `.update()` the queryset, and faster too. Let the DB do the work

Reply:
Didn't know about that, thx :)